src/visitor.rs

```diff
@@ -775,10 +775,10 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
         for macro_selector in config.skip_macro_invocations().0 {
             match macro_selector {
                 MacroSelector::Name(name) => macro_names.push(name.to_string()),
-                MacroSelector::All => skip_context.all_macros = true,
+                MacroSelector::All => skip_context.macros.set_all(true),
             }
         }
-        skip_context.update_macros(macro_names);
+        skip_context.macros.append(macro_names);
```

Comment:
Along with the comments above, I wonder if it makes sense to avoid calling `to_string` when "*" is in the `skip_macro_invocations` list by making the following update: 

```rust
let contains_all = config.skip_macro_invocations().0.any(MacroSelector::is_all);

if contains_all {
    skip_context.macros.set_all(true);
} else {
    skip_context.macros.append(config.skip_macro_invocations().0.collect());
}
```

The code might need to be written a little differently, but I hope the general idea is understood.

Reply:
I think realistically it's unlikely we see a case that makes it worth this optimization, at the cost of reading the vec twice. My expectation is that `"*"`, if given, will be the only string in the vec, as it makes all other names redundant. The unit tests that check combinations of the two are just for completeness.